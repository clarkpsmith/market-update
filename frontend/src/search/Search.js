import React from "react";
import "./Search.css";
import SearchBlock from "../searchBlock/SearchBlock";

const Search = () => {
  return (
    <div className="Search">
      <SearchBlock />
    </div>
  );
};

export default Search;
