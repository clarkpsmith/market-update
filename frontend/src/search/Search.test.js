import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search";
import { Provider } from "react-redux";

import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();

const store = mockStore({
  currentUser: {
    firstName: "test",
    lastName: "user",
    email: "testuser@gmail.com",
    username: "testuser",
    watchlist: ["AAPL"],
  },
});
it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <Search />
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Search />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
