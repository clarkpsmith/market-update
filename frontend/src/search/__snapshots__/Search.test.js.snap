// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <section
      class="SearchBlock"
    >
      <div
        class="SearchBlock-card"
      >
        <div
          class="SearchBlock-search"
        >
          <div
            class="SearchForm"
          >
            <form
              class="SearchForm-form form-inline"
            >
              <input
                class="SearchForm-input"
                id="searchTerm"
                name="searchTerm"
                placeholder="Enter ticker symbol..."
                type="text"
                value=""
              />
              <button
                class="SearchForm-btn btn btn-secondary btn-md"
              >
                Search
              </button>
            </form>
          </div>
        </div>
        <div
          class="results"
        />
        <div
          class="SearchBlock-no-results"
        />
      </div>
    </section>
  </div>
</DocumentFragment>
`;
